Mathlib/LinearAlgebra/Matrix/Spectrum.lean

```diff
@@ -53,139 +50,126 @@ noncomputable def eigenvectorBasis : OrthonormalBasis n 𝕜 (EuclideanSpace 
-theorem spectral_theorem :
-    hA.eigenvectorMatrixInv * A = diagonal ((↑) ∘ hA.eigenvalues) * hA.eigenvectorMatrixInv := by
-  rw [eigenvectorMatrixInv, PiLp.basis_toMatrix_basisFun_mul]
-  ext i j
-  have := isHermitian_iff_isSymmetric.1 hA
-  convert this.eigenvectorBasis_apply_self_apply finrank_euclideanSpace (EuclideanSpace.single j 1)
-    ((Fintype.equivOfCardEq (Fintype.card_fin _)).symm i) using 1
-  · dsimp only [EuclideanSpace.single, toEuclideanLin_piLp_equiv_symm, toLin'_apply,
-      Matrix.of_apply, IsHermitian.eigenvectorBasis]
-    simp_rw [mulVec_single, mul_one, OrthonormalBasis.coe_toBasis_repr_apply,
-      OrthonormalBasis.repr_reindex]
-    rfl
-  · simp only [diagonal_mul, (· ∘ ·), eigenvalues]
-    rw [eigenvectorBasis, Basis.toMatrix_apply, OrthonormalBasis.coe_toBasis_repr_apply,
-      OrthonormalBasis.repr_reindex, eigenvalues₀, PiLp.basisFun_apply, WithLp.equiv_symm_single]
-#align matrix.is_hermitian.spectral_theorem Matrix.IsHermitian.spectral_theorem
-
-theorem eigenvalues_eq (i : n) :
-    hA.eigenvalues i =
-      RCLike.re (star (hA.eigenvectorMatrixᵀ i) ⬝ᵥ A *ᵥ hA.eigenvectorMatrixᵀ i) := by
-  have := hA.spectral_theorem
-  rw [← @Matrix.mul_inv_eq_iff_eq_mul_of_invertible (A := hA.eigenvectorMatrixInv)] at this
-  have := congr_arg RCLike.re (congr_fun (congr_fun this i) i)
-  rw [diagonal_apply_eq, Function.comp_apply, RCLike.ofReal_re,
-    inv_eq_left_inv hA.eigenvectorMatrix_mul_inv, ← conjTranspose_eigenvectorMatrix, mul_mul_apply]
-    at this
-  exact this.symm
-#align matrix.is_hermitian.eigenvalues_eq Matrix.IsHermitian.eigenvalues_eq
-
-/-- The determinant of a hermitian matrix is the product of its eigenvalues. -/
-theorem det_eq_prod_eigenvalues : det A = ∏ i, (hA.eigenvalues i : 𝕜) := by
-  apply mul_left_cancel₀ (det_ne_zero_of_left_inverse (eigenvectorMatrix_mul_inv hA))
-  rw [← det_mul, spectral_theorem, det_mul, mul_comm, det_diagonal]
-  simp_rw [Function.comp_apply]
-#align matrix.is_hermitian.det_eq_prod_eigenvalues Matrix.IsHermitian.det_eq_prod_eigenvalues
-
-/-- *spectral theorem* (Alternate form for convenience) A hermitian matrix can be can be
-replaced by a diagonal matrix sandwiched between the eigenvector matrices. This alternate form
+theorem star_mul_self_mul_eq_diagonal :
+    (star (eigenvectorUnitary hA : Matrix n n 𝕜)) * A * (eigenvectorUnitary hA : Matrix n n 𝕜)
+      = diagonal (RCLike.ofReal ∘ hA.eigenvalues) := by
+apply Matrix.toEuclideanLin.injective
+apply Basis.ext (EuclideanSpace.basisFun n 𝕜).toBasis
+intro i
+simp only [toEuclideanLin_apply, OrthonormalBasis.coe_toBasis, EuclideanSpace.basisFun_apply,
+ WithLp.equiv_single, ← mulVec_mulVec, eigenvectorUnitary_mulVec, ← mulVec_mulVec,
+ mulVec_eigenvectorBasis, Matrix.diagonal_mulVec_single, mulVec_smul,
+ star_eigenvectorUnitary_mulVec, RCLike.real_smul_eq_coe_smul (K := 𝕜), WithLp.equiv_symm_smul,
+ WithLp.equiv_symm_single, Function.comp_apply, mul_one, WithLp.equiv_symm_single]
```

Comment:
Sorry, I think I screwed up in my previous style fix here. I shouldn't make GitHub suggestions from my phone.
```suggestion
  WithLp.equiv_single, ← mulVec_mulVec, eigenvectorUnitary_mulVec, ← mulVec_mulVec,
  mulVec_eigenvectorBasis, Matrix.diagonal_mulVec_single, mulVec_smul,
  star_eigenvectorUnitary_mulVec, RCLike.real_smul_eq_coe_smul (K := 𝕜), WithLp.equiv_symm_smul,
  WithLp.equiv_symm_single, Function.comp_apply, mul_one, WithLp.equiv_symm_single]
```